src/lib/sqle/spooled-query.service.ts

```diff
@@ -0,0 +1,130 @@
+import { Injectable, Provider, Optional, SkipSelf } from '@angular/core';
+import { HttpErrorResponse } from '@angular/common/http';
+import { TranslateService } from '@ngx-translate/core';
+import { catchError, expand, map, mapTo, skipWhile, switchMap, take, tap } from 'rxjs/operators';
+import { BehaviorSubject, Observable, of, throwError, timer } from 'rxjs';
+import { VantageConnectionService } from './connection.service';
+import { VantageQueryService, IQueryPayload, IQueryResultSet } from './query.service';
+
+interface ISpooledQueryError extends HttpErrorResponse {
+  id: string;
+  detailMessage: string;
+}
+
+const MAX_INTERVAL: number = 10000;
+const BASE_INTERVAL: number = 2000;
+const MAX_RETRIES: number = 2;
+
+enum SpooledQueryState {
+  QUEUED = 'QUEUED',
+  PENDING = 'PENDING',
+  SUBMITTED = 'SUBMITTED',
+  RESPONDING = 'RESPONDING',
+  SPOOLING = 'SPOOLING',
+  RESULT_SET_READY = 'RESULT_SET_READY',
+}
+
+@Injectable()
+export class VantageSpooledQueryService {
+  queryStatus: BehaviorSubject<SpooledQueryState> = new BehaviorSubject<SpooledQueryState>(undefined);
+  queryStatus$: Observable<SpooledQueryState> = this.queryStatus.asObservable();
```

Comment:
We might need to do a map for this to have an observable per execution. Thoughts?

I see potential issues when trying to run 2 queries at the same time.

Reply:
That makes sense. We can fix it by changing `querySystem` to follow the pattern in Oseer's example up above, but that's probably for another PR.